### PR TITLE
Change way of loading template to not base on plug-in start method

### PR DESCRIPTION
**Pull request by hsd99**
[JENKINS-27936] When few FitNesse test results being published in a single job they renamed into Summary

### CODE REVIEW

File: src/main/java/hudson/plugins/fitnesse/FitnessePlugin.java [hunk 1]
```diff
@@ -16,16 +16,21 @@
...
-	public static Transformer newRawResultsTransformer() throws TransformerException {
+	public static Transformer newRawResultsTransformer() throws TransformerException, IOException {
+		if (templates != null)
+			return templates.newTransformer();
+
+		TransformerFactory transformerFactory = TransformerFactory.newInstance();
+		StreamSource xslSource = new StreamSource(getXslAsInputStream());
+		templates = transformerFactory.newTemplates(xslSource);
 		return templates.newTransformer();
 	}
```
Comment (antoine-aumjaud) [inline src/main/java/hudson/plugins/fitnesse/FitnessePlugin.java]: If plugin life cycle is not used anymore, remove this class and move the content to the clash which use newRawResultsTransformer

File: src/main/java/hudson/plugins/fitnesse/FitnesseResults.java [hunk 2]
```diff
@@ -391,22 +406,24 @@ public boolean hasChildren() {
...
 	}
 
 	/**
-	 * Returns <code>true</code> if this results has html content
-	 * that is available via {@link #getHtmlContent()}
+	 * Returns <code>true</code> if this results has html content that is
+	 * available via {@link #getHtmlContent()}
 	 */
 	protected boolean hasHtmlContent() {
 		return pageCounts != null && pageCounts.contentFile != null;
```
Comment (antoine-aumjaud) [inline src/main/java/hudson/plugins/fitnesse/FitnesseResults.java]: Format only the code you have modify, it's easier to find what you have changed

File: src/main/java/hudson/plugins/fitnesse/NativePageCounts.java [hunk 3]
```diff
@@ -59,7 +61,7 @@ public void startElement(String uri, String localName, String qName, Attributes
 		if (COUNTABLE.contains(qName)) {
 			String targetPage;
 			if (qName.equals(SUMMARY)) {
-				targetPage = "Summary";
+				targetPage = resultFileName;
 			} else {
 				String page = attributes.getValue(PAGE);
 				String pseudoPage = attributes.getValue(PSEUDO_PAGE);
```
Comment (antoine-aumjaud) [inline src/main/java/hudson/plugins/fitnesse/NativePageCounts.java]: this commit concerne the bug of JENKINS-27936, please do 2 differents commits, and 2 differents pull requests

**Comment by antoine-aumjaud** [comment]
Thanks for your PR Pawel, could you do 2 PR ?

File: src/main/java/hudson/plugins/fitnesse/FitnessePlugin.java (hunk 1 above)
Comment (hsd99) [inline src/main/java/hudson/plugins/fitnesse/FitnessePlugin.java]: I can move it. I'm only not sure what happens when eliminate FitnessePlugin which inhered from Jenkins Plugin.

**Comment by hsd99** [comment]
Thanks for suggestion. I will do two RP.

**Pull request closed without merging**